timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -870,6 +854,9 @@ def assign(self, **kwargs) -> TimeSeriesDataFrame:
         """Assign new columns to the time series dataframe. See :meth:`pandas.DataFrame.assign` for details."""
         return super().assign(**kwargs)  # type: ignore
 
+    def sort_index(self, *args, **kwargs) -> TimeSeriesDataFrame:
```

Comment:
Maybe add a comment explaining that this method was only added for pyright? Similarly for other methods following this pattern.